src/services/submissionService.ts

```diff
@@ -118,7 +118,23 @@ function convertToDbSubmission(submission: gqlType.SubmissionInput):
 export const addSubmission = async (submissionInput: gqlType.SubmissionInput): 
     Promise<dbSchema.Submission> => {
     try {
-        const dbSubmission = convertToDbSubmission(submissionInput)
+        if (!submissionInput.googleMapsUrl.trim() || !submissionInput.healthcareProfessionalName.trim()) {
+            throw new Error('Submission fields cannot be empty strings.')
+        }
+
+        const nonNullLanguages = submissionInput.spokenLanguages
+            .filter(lang => lang !== null) as gqlType.SpokenLanguageInput[]
+
+        const validatedLanguages = mapAndValidateSpokenLanguages(nonNullLanguages)
+
+        if (!validatedLanguages || validatedLanguages.length === 0) {
```

Comment:
Move this into the map and validate function. 
This is probably a good time for us to introduce a proper errors object. 
Have it return a `hasErrors` Boolean and and errors array where each error is an object with a `message` field and `property` field
Example

```
result: {
    hasErrors: true,
    errors: [
        { field: "spokenlanguages[1].type",
          message: "the type can only be english or Japanese" }]
}
```

Reply:
update: 
1) the error message should be based on the custom errors type.
2) we should make this result object a typscript type that can be used across all the services.

Reply:
update #2: I have a PR with a results framework set up that should make this a bit easier and more consistent.